pkg/dependencies/factory.go

```diff
@@ -328,15 +328,24 @@ func (f *Factory) WithTroubleshoot() *Factory {
 }
 
 func (f *Factory) WithNetworking(clusterConfig *v1alpha1.Cluster) *Factory {
+	var networkingBuilder func() clustermanager.Networking
+	if clusterConfig.Spec.ClusterNetwork.CNI == v1alpha1.Kindnetd {
```

Comment:
What are we gaining from refactoring this instead of just include the kubectl build under the Kindnetd condition like before?

Reply:
I'm no sure I understand the question
The previous condition was executed during "build" time
We need to define the dependency on `kubectl` before that, so `kubectl` gets built before `kindnet`

Reply:
I get it now after discussing, looks good